intellij.md

```diff
@@ -0,0 +1,107 @@
+
+Beide Versionen lassen sich von der [Produktseite](https://www.jetbrains.com/de-de/idea/) herunterladen.
+Sollten Sie sich für die Ultimate Version entscheiden, ist eine Registrierung und Bestätigung ihres Studierendenstatus erforderlich. 
+Folgen Sie dazu den entsprechenden Anweisungen im Abschnitt [Benutzererstellung und Aktivierung](#benutzererstellung-und-Aktivierung). 
+
+## Benutzererstellung und Aktivierung
+*Dieser Abschnitt ist nur relevant, wenn Sie die Ultimate Version nutzen möchten.*
+
+Um die Ultimate Version nutzen zu können, benötigen Sie einen JetBrains Account und müssen diesen mit Ihrem Studierendenstatus verknüpfen.
+Die Verknüpfung findet dabei über GitHub statt.
+
+1. Öffnen Sie [GitHub](https://github.com/) und melden Sie sich mit Ihrem Account an.
+2. Öffnen Sie Ihre Profileinstellungen und navigieren Sie zu `Emails`.
+3. Fügen Sie hier ihre Email-Adresse der Hochschule Flensburg hinzu (endet auf @stud.hs-flensburg.de).
+4. Anschließend bekommen Sie von GitHub eine Bestätigungs-Email zugeschickt. Öffnen Sie diese und klicken Sie auf `Confirm email address`.
+5. Öffnen Sie die GitHubs [Sudierendenportal](https://education.github.com/) und melden Sie sich mit Ihrem GitHub Account an.
+6. Klicken Sie auf `Join Global Campus` und anschließend auf `Yes, I'm a student`. 
+7. Wählen Sie in dem untenstehenden Formular Ihre Hochschul-Email-Adresse aus, schreiben Sie einen kurzen Text in das folgende Textfeld und klicken Sie auf `Continue`.
+8. Fügen Sie ein Bild Ihrer Studierendenbescheiniung bei und klicken Sie auf `Process my application`.
+    - Die Verifizierung durch GitHub dauert in der Regel nur wenige Minuten.
+9. Öffnen Sie nun das JetBrains [Studierendenportal](https://www.jetbrains.com/shop/eform/students).
+10. Klicke Sie auf `GitHub` und anschließend auf `Mit GitHub autorisieren`.
+11. Melden Sie sich mit Ihrem GitHub Account an und klicken Sie auf `Authorize`.
+12. Folgen Sie den Anweisungen des Formulars und klicken Sie auf `Kostenlose Produkte beantragen`
+13. Über die [Lizenzverwaltung](https://account.jetbrains.com/licenses) können Sie nun Ihr `JetBrains Product Pack for Students` aktivieren und alle Produkte von JetBrains herunterladen.
+
+## Import eines Projekts
+Den Import eines GitHub Projekts können Sie entweder manuell vornehmen oder über die GitHub Integration in IntelliJ durchführen.
+
+### Manuell
+1. Clonen Sie das Projekt in einen beliebigen Ordner auf Ihrem Rechner.
+2. Öffnen Sie IntelliJ und wählen Sie `File > New > Project from Existing Sources...`.
+3. Navigieren Sie zu dem Ordner, in dem Sie das Projekt abgelegt haben und wählen Sie diesen aus.
+4. Klicken Sie auf `Ok`.
+
+### GitHub Integration
+Einmalig: Navigieren Sie in IntelliJ zu `File > Settings > Version Control > GitHub` und setzen Sie den Haken bei `Clone git repositories using ssh`.
+1. Öffnen Sie IntelliJ und wählen Sie `File > New > Project from Version Control...`.
```

Comment:
Es wird jetzt ja für jede Aufgabe ein neues Projekt erzeugt. Grundsätzlich könnte man die Aufgaben auch als Modul hinzufügen, so dass man alle Aufgaben in einem Workspace hat. In diesem Fall ist das Kopieren des existierenden Codes etwas einfacher. Wie stehst du dazu?

Reply:
Verstehe ich es richtig, dass du das bestehende Repository automatisch um Module ergänzen kannst, die dann nur gepullt werden müssen?

Wenn die neuen Module dann automatisch auf die zuvor angelegten Klassen zugreifen würden und dann diese fortlaufend angepasst werden, finde ich die Idee sehr gut. Dies würde auf jeden Fall die Hürde nehmen zu schauen, welche Klassen wann relevant sind, neu angelegt werden müssen oder von vorherigen Aufgaben weiterverwendet werden können.

Reply:
Ne, so wie du es beschrieben hast, öffnet man in IntelliJ jetzt jedes mal ein neues Projekt. Das heißt, man sieht nur genau die Klassen, die zur aktuellen Aufgabe gehören. Man kann ein Gradle-Projekt aber auch als Modul importieren. Dann hast du, ähnlich wie bei Eclipse einfach mehrere Java-Projekte in deinem Workspace. Aus meiner Sicht ist das ganz sinnvoll, da man dann Klassen von einem Java-Projekt zum anderen verschieben können sollte. Magst du dir einmal durchlesen, wie die Module bei IntelliJ funktionieren?

Reply:
Ich habe gelesen, dass die Module in IntelliJ analog zu Eclipse umgesetzt sind. 

Zum testen habe ich: 
1. Ein neues Projekt `Algorithmen` angelegt
2. Einige Repositories aus Git in ein separates Verzeichnis gecloned
3. Die Projekt-Struktur unter `File > Project Structure...` geöffnet
4. Im Seitenmenü `Modules` ausgewählt 
5. Die Repositories über `+ > Import Module` als Module hinzugefügt

Auf Punkt 1 kann auch verzichtet werden, sodass zunächst z.B. Labor 01 wie gewohnt als Projekt angelegt wird und die weiteren Laboraufgaben dann als Modul hinzugefügt werden. 

Alternativ kann man zur obigen Abfolge auch direkt über `File > New > Module from Existing Sources...` gehen. 

Ich habe aber leider keine Möglichkeit gefunden um Module über VCS anzulegen und direkt von Git zu clonen.

Reply:
Ja, okay, dann lass doch die Beschreibung einfach so wie sie ist. Für die Studierenden ist es vielleicht in Ordnung, für jede Aufgabe ein eigenes Projekt zu haben. Sie müssen die Klassen, die sie weiterverwenden wollen dann halt auf Dateiebene kopieren.